### ISSUE TITLE
Error in 22.1-1 

### ISSUE THREAD

**Issue by zjplab** [gzc/CLRS#198]
check [stackoverflow](https://stackoverflow.com/questions/18071576/adjacency-list-representation-of-a-directed-graph)

**Comment by mhhacker36** [comment]
I agree with you @zjplab . My approach is adding out-degree and in-degree attributes to each vertices just like PI and D attributes and calculate the value while searching the Graphs only. So we need to initialize these attributes just like we initialize PI and D attributes value. So that's O(V).
And searching part of BFS takes O(E) time. Hence total of O(V+E). Hope this helps.
